staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/controller/controller.go

```diff
@@ -270,7 +270,7 @@ func (d *DynamicEncryptionConfigContent) validateNewTransformersHealth(
 	ctx, cancel = context.WithTimeout(ctx, kmsPluginCloseGracePeriod)
 	defer cancel()
 
-	pollErr := wait.PollImmediateWithContext(ctx, 100*time.Millisecond, kmsPluginCloseGracePeriod, func(ctx context.Context) (bool, error) {
+	pollErr := wait.PollUntilContextTimeout(ctx, 100*time.Millisecond, kmsPluginCloseGracePeriod, false, func(ctx context.Context) (bool, error) {
```

Comment:
... shouldn't this be `true`?

Reply:
It seems that the logic is correct. Let me make some modifications

Reply:
Immediate means you need to use true here, otherwise you are modifying the behavior